storey/aggregations.py

```diff
@@ -99,18 +105,66 @@ async def _emit_worker(self):
             next_emit_time = next_emit_time + seconds_to_sleep_between_emits
 
 
+# Serving side for AggregateByKey, for every (key, timestamp) pair provided QueryAggregateByKey will emit the relevant feature set
+class QueryAggregationByKey(AggregateByKey):
+    def __init__(self, aggregates, cache, key=None, emit_policy=_default_emit_policy, augmentation_fn=None):
+        AggregateByKey.__init__(self, aggregates, cache, key, emit_policy, augmentation_fn)
+        self._aggregates_store.read_only = True
+
+    async def _do(self, event):
+        if event == _termination_obj:
+            self._terminate_worker = True
+            return await self._do_downstream(_termination_obj)
+
+        # check whether a background loop is needed, if so create start one
+        if (not self._emit_worker_running) and \
+                (isinstance(self._emit_policy, EmitAfterPeriod) or isinstance(self._emit_policy, EmitAfterWindow)):
+            asyncio.get_running_loop().create_task(self._emit_worker())
+            self._emit_worker_running = True
+
+        element = event.body
+        key = event.key
+        if self.key_extractor:
+            key = self.key_extractor(element)
+
+        if isinstance(self._emit_policy, EmitEveryEvent):
+            await self._emit_event(key, event)
+        elif isinstance(self._emit_policy, EmitAfterMaxEvent):
+            self._events_in_batch[key] = self._events_in_batch.get(key, 0) + 1
+            if self._events_in_batch[key] == self._emit_policy.max_events:
+                await self._emit_event(key, event)
+                self._events_in_batch[key] = 0
+
+
+class Persist(Flow):
+    def __init__(self, cache):
+        super().__init__()
+        self._cache = cache
+
+    async def _do(self, event):
+        if event is _termination_obj:
+            return await self._do_downstream(_termination_obj)
+        else:
+            await self._cache.persist_key(event.key)
```

Comment:
This is going to persist sequentially, which we may be willing to accept here as a first step, but is probably important enough to warrant a TODO.

Reply:
You're right, i added a todo for now